src/quick-lint-js/diag/diagnostic-types-2.h

```diff
@@ -2833,6 +2833,33 @@ struct Diag_Unclosed_Code_Block {
   Source_Code_Span block_open;
 };
 
+struct Diag_Mismatched_Curly_Indentation {
+  [[qljs::diag("E0441", Diagnostic_Severity::warning)]]  //
+  [[qljs::message("indentation level of '{0}' does not match '{1}' below",
+                  ARG(left_curly), ARG(right_curly))]]  //
+  [[qljs::message("indentation level of '{0}' does not match '{1}' above",
+                  ARG(right_curly), ARG(left_curly))]]  //
+  Source_Code_Span left_curly;
+  Source_Code_Span right_curly;
+};
+
+struct Diag_Misleading_Braceless_If_Else_Indentation {
+  [[qljs::diag("E0442", Diagnostic_Severity::warning)]]  //
+  [[qljs::message("indentation of '{0}' does not match '{1}'", ARG(if_span),
+                  ARG(else_span))]]  //
+  [[qljs::message("indentation of '{0}' does not match '{1}'", ARG(else_span),
+                  ARG(if_span))]]  //
+  Source_Code_Span if_span;
+  Source_Code_Span else_span;
+};
+
+struct Diag_Misleading_If_Or_Else_Body_Indentation {
+  [[qljs::diag("E0443", Diagnostic_Severity::warning)]]  //
+  [[qljs::message("misleading indentation after 'if' body",
+                  ARG(if_or_else_span))]]  //
+  Source_Code_Span if_or_else_span;
```

Comment:
I think this is the wrong spot. We should point at the following statement or at the indentation before the following statement. The location of the `if` or `else` is supplementary information.
```javascript
  if (cond)
      f();
      g();
//^^^^ warning: misleading indentation after 'if' body
```